### PR TITLE
[SPARK-45351][CORE] Change `spark.shuffle.service.db.backend` default value to ROCKSDB

### PR DESCRIPTION
**Pull request by Hisoka-X**
### What changes were proposed in this pull request?
Because spark will remove LevelDB in the future. So this PR change `spark.shuffle.service.db.backend` default value to ROCKSDB, not LEVELDB.

### Why are the changes needed?
Prepare remove leveldb.

### Does this PR introduce _any_ user-facing change?
Yes, RocksDB will as default value of `spark.shuffle.service.db.backend`

### How was this patch tested?
exist test.

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

**Comment by LuciferYang** [comment]
https://github.com/apache/spark/blob/eff46ea77e9bebe7777f3076277bef1e086833dd/core/src/main/scala/org/apache/spark/internal/config/package.scala#L726-L734

This place also needs to be fixed, and there should also be places in the document that need to be changed.

**Comment by LuciferYang** [comment]
And this part in `network-yarn`

https://github.com/apache/spark/blob/eff46ea77e9bebe7777f3076277bef1e086833dd/common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java#L260-L266

**Review by LuciferYang — APPROVED** [review]
+1, LGTM if test pass

**Review by mridulm — COMMENTED** [review]
Thanks for working on this @Hisoka-X ! 

A few additional changes which should help - some of these should have been done when we introduced `rockdb` :-(
* Rename `ShuffleTestAccessor.mergeManagerLevelDB` to `ShuffleTestAccessor.mergeManagerDB`
* Some of the comments would be outdated with this change. For example, [`FSHistoryProvider.loadDiskStore`](https://github.com/apache/spark/blob/187e9a851758c0e9cec11edab2bc07d6f4404001/core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala#L1228), `HybridStore` ([1](https://github.com/apache/spark/blob/187e9a851758c0e9cec11edab2bc07d6f4404001/core/src/main/scala/org/apache/spark/deploy/history/HybridStore.scala#L45), [2](https://github.com/apache/spark/blob/187e9a851758c0e9cec11edab2bc07d6f4404001/core/src/main/scala/org/apache/spark/deploy/history/HybridStore.scala#L52)), etc.
* Not sure if this is still used, but `HistoryServerSuite.main` uses LevelDB.
* Some of the exception and error messages are still using level db - example in `HybridStore`, `ExternalShuffleBlockResolver`, etc.

The list above is more indicative than exhaustive, please do take a look at other examples as well.

**Review by dongjoon-hyun — COMMENTED** [review]
+1 for Mridulm 's advice.

**Comment by LuciferYang** [comment]
+1 for Mridulm 's advice.

File: common/kvstore/src/main/java/org/apache/spark/util/kvstore/RocksDBIterator.java [hunk 1]
```diff
@@ -255,8 +255,8 @@ static boolean startsWith(byte[] key, byte[] prefix) {
 
   private boolean isEndMarker(byte[] key) {
     return (key.length > 2 &&
-        key[key.length - 2] == LevelDBTypeInfo.KEY_SEPARATOR &&
-        key[key.length - 1] == LevelDBTypeInfo.END_MARKER[0]);
+        key[key.length - 2] == RocksDBTypeInfo.KEY_SEPARATOR &&
+        key[key.length - 1] == RocksDBTypeInfo.END_MARKER[0]);
```
[the remark this answers is not in the document]
Comment (mridulm) [reply common/kvstore/src/main/java/org/apache/spark/util/kvstore/RocksDBIterator.java:259]: Nice !

**Review by mridulm — APPROVED** [review]
Looks good to me.
+CC @LuciferYang, @dongjoon-hyun

**Review by LuciferYang — APPROVED** [review]
+1, LGTM

**Comment by mridulm** [comment]
Thanks for reviewing @LuciferYang !
Can you merge it please ? I am AFK :-)

**Comment by LuciferYang** [comment]
Merged into master for Spark 4.0. Thanks @Hisoka-X @mridulm and @dongjoon-hyun  ~

**Pull request closed without merging**